### ISSUE TITLE
In driveFieldCentric - if you xSpeed and ySpeed 0 you cannot rotate

### ISSUE THREAD

**Issue by foocross** [FTCLib/FTCLib#81]
**Describe the bug**
When driveFieldCentric is called and xSpeed and ySpeed are 0, rotation does not work. So if you are only trying to rotate, the robot will not move. We think the problem is at:
        normalize(wheelSpeeds, input.magnitude());
input is built from:
Vector2d input = new Vector2d(xSpeed, ySpeed);

which means that the magnitude should be 0 if both x and y are 0.

Normalize should check to see if rotation is also 0 or not....

We are not sure what else would happen if we replaced normalize with:

if (input.magnitude() != 0) { // only normalize when we are moving x,y..
  normalize(wheelSpeeds, input.magnitude());
} 

**To Reproduce**
Try to rotate the robot when xSpeed and ySpeed are 0.

**Expected behavior**
The robot should rotate.

**Additional context**
Add any other context about the problem here. We would recommend telling us yout:
version 1.0.1

**Comment by JIceberg** [comment]
That's not possible.
```java
double[] wheelSpeeds = new double[4];
        wheelSpeeds[MotorType.kFrontLeft.value] =
                input.magnitude() * Math.sin(theta + Math.PI / 4) + turnSpeed;
        wheelSpeeds[MotorType.kFrontRight.value] =
                input.magnitude() * Math.sin(theta - Math.PI / 4) - turnSpeed;
        wheelSpeeds[MotorType.kBackLeft.value] =
                input.magnitude() * Math.sin(theta - Math.PI / 4) + turnSpeed;
        wheelSpeeds[MotorType.kBackRight.value] =
                input.magnitude() * Math.sin(theta + Math.PI / 4) - turnSpeed;
```
Your wheel speed is also determined by the turnSpeed input. This is what turns your robot.

**Comment by JIceberg** [comment]
Ah I see what you mean. The normalization is multiplying it by 0 when the input magnitude is 0. This will be fixed.

**Comment by JIceberg** [comment]
So, I see where the issue is, I believe.
```java
protected void normalize(double[] wheelSpeeds, double magnitude) {
        double maxMagnitude = Math.abs(wheelSpeeds[0]);
        for (int i = 1; i < wheelSpeeds.length; i++) {
            double temp = Math.abs(wheelSpeeds[i]);
            if (maxMagnitude < temp) {
                maxMagnitude = temp;
            }
        }
        for (int i = 0; i < wheelSpeeds.length; i++) {
            wheelSpeeds[i] = (wheelSpeeds[i] / maxMagnitude) * magnitude;
        }

    }
```
I think you are correct with the input magnitude affecting the resulting speed as it is giving 0. This will be fixed. Thank you!

**Comment by JIceberg** [comment]
I believe the issue should be resolved in 1.0.1, I updated the files in bintray. Let me know how it works.

**Comment by JIceberg** [comment]
Issue will be closed due to inactivity in 3 days. Are there any other problems you've come across? @foocross 
The implementation has also been updated for v1.1.0 to be a bit smoother in terms of the normalization (i.e. not relying on translation magnitude for all movement).
